src/ac_power_flow_residual.jl

```diff
@@ -276,27 +290,33 @@ function _update_residual_values!(
     Q_net::Vector{Float64},
     P_net_set::Vector{Float64},
     bus_slack_participation_factors::SparseVector{Float64, Int},
-    slack_bus::Int,
+    subnetworks::Dict{Int64, Vector{Int64}},
     data::ACPowerFlowData,
     time_step::Int64,
 )
     # update P_net, Q_net, data.bus_angles, data.bus_magnitude based on X
     Yb = data.power_network_matrix.data
     bus_types = view(data.bus_type, :, time_step)
-    P_slack =
-        (x[2 * slack_bus - 1] - P_net_set[slack_bus]) .* bus_slack_participation_factors
-    for (ix, bt) in enumerate(bus_types)
-        _set_state_vars_at_bus!(
-            ix,
-            P_net,
-            Q_net,
-            P_net_set,
-            P_slack[ix],
-            x,
-            data,
-            time_step,
-            Val(bt),
-        )
+
+    for (slack_bus, subnetwork_buses) in subnetworks
```

Comment:
This isn't looping through all `ACBusTypes.REF` buses `slack_bus`, just the `first` one in each subnetwork, right? Is that arbitrariness a problem?